docs/DeveloperGuide.md

```diff
@@ -481,6 +481,201 @@ Below is an Activity Diagram illustrating the flow of activities when the applic
+If there are missing or invalid keywords, a `ParseException` will be thrown.
+
+The `parse` method takes in a String of user input, `args` and contains a `ArgumentMultimap` object, `argMultimap`.
+Regular expressions are used to identify whether optional keywords like `remind` and `/p` are present.
+The remind keyword is identified using `.*\bremind\b.*` while priority keyword is identified using `.*\bp/\b.*`.
+In the event that the remind keyword is present, `remind` will be removed from `args` before the parsing of the other keywords by `argMultimap`.
+The `tokenize` method of `ArgumentTokenizer` will be called. The keywords are parsed and return as `argMultimap`.
+The keywords are subsequently extracted from `argMultimap` to create a new `Assignment` object, `assignment`, which is used
+to return a new `AddCommand` object. 
+ 
+In the event that the keywords have an invalid format, such as the `Deadline` keyword not being in the required `dd-MM-yyyy HHmm` form for example,
+a `ParseException` will be thrown. 
+
+#### Usage Scenario
+The following is a usage scenario when a user wants to add an assignment with the name 'Lab', module 'CS2103', and deadline '10-10-2020 2359'.
+
+1. `execute ("add n/Lab mod/CS2103 d/10-10-2020 2359")` of `LogicManager` calls the `parseCommand` method of `ProductiveNusParser`.
+2. `parseCommand("add n/Lab mod/CS2103 d/10-10-2020 2359")` parses the String `"add n/Lab mod/CS2103 d/10-10-2020 2359"` and returns an initialized `AddCommandParser` object.
+3. `parseCommand("add n/Lab mod/CS2103 d/10-10-2020 2359")` calls the `parse` method in `AddCommandParser` which parses the user input into `Name`, `Time` and `ModuleCode`.
+ This is done by calling the methods `tokenize` followed by `getValue` of `ArgumentMultimap`.
+4. If user input is valid, a `Assignment` object will be created, and it will be used to return a `AddCommand` object.
+5. There is a return call to `LogicManager` which then calls the overridden `execute` method of `AddCommand`.
+6. The `execute` method then calls `hasAssignment` of `model` to check if the `toAdd` assignment is a duplicate.
+7. If the assignment is not a duplicate, the `execute` method of `AddCommand` will call the `addAssignment` method of the `Model` object.
+8. The `execute` method returns a `CommandResult` object.
+
+Given below is the sequence diagram for the interactions within `LogicManager` for the `execute ("add n/Lab mod/CS2103 d/10-10-2020 2359")` API call.
+
+   ![Sequence Diagram for AddCommand](images/AddSequenceDiagram.png)
+   <br/>*Figure X: Sequence Diagram for AddCommand*
+
+### \[Implemented\] Marking assignments as done and Setting reminders for assignments features
+
+Both features are implemented in a similar way, though the reasons for implementation differs.There are also some differences in the implementation, which will be pointed out along the way.
```

Comment:
Still has the implemented thingy in the header

Reply:
fixed!